kornia/augmentation/random_generator/random_generator.py

```diff
@@ -170,7 +170,7 @@ def random_affine_generator(
     # compute tensor ranges
     if scale is not None:
         _joint_range_check(cast(torch.Tensor, scale[:2]), "scale")
-        scale = _adapted_uniform((batch_size,), scale[0], scale[1], same_on_batch).repeat(1, 2)
+        scale = _adapted_uniform((batch_size,), scale[0], scale[1], same_on_batch).unsqueeze(1).repeat(1, 2)
```

Comment:
why any test were crashing ?

Reply:
It will crash when batchsize > 1 and len(scale) > 1. Our tests is not that thorough. We need to test more edge cases.

Reply:
add some tests here or in another PR